components/sidebar/sidebar_channel_list/sidebar_channel_list.tsx

```diff
@@ -288,12 +291,27 @@ export default class SidebarChannelList extends React.PureComponent<Props, State
         return this.getFirstUnreadChannelFromChannelIdArray(this.getDisplayedChannelIds().reverse());
     }
 
+    navigateByChannelId = (id: string) => {
+        if (this.props.collapsedThreads && id === '') {
+            this.props.actions.switchToGlobalThreads();
+        } else {
+            this.props.actions.switchToChannelById(id);
+        }
+    }
+
     navigateChannelShortcut = (e: KeyboardEvent) => {
         if (e.altKey && !e.shiftKey && !e.ctrlKey && !e.metaKey && (Utils.isKeyPressed(e, Constants.KeyCodes.UP) || Utils.isKeyPressed(e, Constants.KeyCodes.DOWN))) {
             e.preventDefault();
 
             const allChannelIds = this.getDisplayedChannelIds();
             const curChannelId = this.props.currentChannelId;
+
+            if (this.props.collapsedThreads) {
+                // threads set channel id to ''
+                // add it to allChannelIds
+                allChannelIds.unshift('');
```

Comment:
Maybe it would be a bit clearer to have a `global_threads` constant for use in this file instead of just treating the empty string as Global Threads? I could see more non-channel items being added to the sidebar in the future that would need their own IDs for this.

Reply:
+1 for a constant

Also an aside mention that once we get a more stable infrastructure for these non-channel pages, that https://github.com/mattermost/mattermost-webapp/blob/master/components/threading/global_threads/global_threads.tsx#L66-L67 should be removed.

Reply:
@hmhealey, @calebroseland while I agree 100% with what you are both saying, 
I don't see how we can use a constant when we don't have a concrete architecture for non-channel sidebar items. 
The issue I see is that we need to check against `currentChannelId` (which is `''`) for global threads.

I'll attempt a commit to support non-channel pages

Reply:
@hmhealey, @calebroseland I've added a commit to address the comments, can you take a look and share your thoughts?